### PR TITLE
Allow captcha when submit a webllink

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for Issue #209 .

#### Summary of Changes
allow to use captcha when submit a weblink 

#### Testing Instructions
use submit weblink enabling/disabling captcha settings

### CODE REVIEW

**Comment by yvesh** [comment]
@alikon Hey Nicola, works fine, but there are some small issues i found so far:

The tooltip text: `Type in the textbox what you see in the image` is not really matching for Recaptcha version 2 (where you don't fill out text boxes, but click images).

Some code style issues (see comments).

Thank you!

File: src/components/com_weblinks/views/form/tmpl/edit.php [hunk 1]
```diff
@@ -14,6 +14,17 @@
 JHtml::_('formbehavior.chosen', 'select');
 JHtml::_('behavior.modal', 'a.modal_jform_contenthistory');
 
+$captchaEnabled = false;
+$captchaSet = $this->params->get('captcha', JFactory::getApplication()->get('captcha', '0'));
+foreach (JPluginHelper::getPlugin('captcha') as $plugin)
```
Comment (yvesh) [inline src/components/com_weblinks/views/form/tmpl/edit.php]: A new empty line is missing before the foreach loop

File: src/components/com_weblinks/views/form/tmpl/edit.php [hunk 2]
```diff
@@ -53,6 +64,12 @@
 			<?php endif; ?>
 		</div>
 
+		<?php if (($captchaEnabled)) : ?>
```
Comment (yvesh) [inline src/components/com_weblinks/views/form/tmpl/edit.php]: Why do you have two braces here?

`if ($captchaEnabled) : `

**Comment by chrisdavenport** [comment]
A couple of issues:
1. If I fail the captcha the page reloads but all the form field entries are lost.
2. Perhaps move the captcha field to the end of the form?

**Comment by alikon** [comment]
@yvesh 

> The tooltip text: Type in the textbox what you see in the image is not really matching for Recaptcha version 

is like com_contact what should be the correct STRING ?

@chrisdavenport 

> 2. Perhaps move the captcha field to the end of the form?

not sure should be better near the submit button ?

> 1. If I fail the captcha the page reloads but all the form field entries are lost.

 any help ?

**Comment by yvesh** [comment]
@alikon Not sure on the text. Something more generic maybe, which fits both.

Please answer the security question. But i am not a native speaker. @chrisdavenport What do you think?

screenshot 2016-06-26 12 13 55

>> If I fail the captcha the page reloads but all the form field entries are lost.

> any help ?

This is not going to be implemented easily. For com_users the data the user entered during registration are saved in the session and then loaded again.

See [this](https://github.com/joomla-projects/gsoc16_browser-automated-tests/blob/staging/components/com_users/controllers/user.php#L337).

**Comment by chrisdavenport** [comment]
I checked with the en-GB team and because "I'm not a robot" is not a question, the suggested text is "Please complete the security check."  There is a PR in the CMS repo for the same text: https://github.com/joomla/joomla-cms/pull/10931

Good point about the submit buttons.  Actually, I think the buttons should be moved to the end too.  See, for example, the contact form.

Should be fairly simple to save the form data in the session as per com_users and also com_contact.

**Comment by alikon** [comment]
+ session form data
+ captcha & buttons moved down
captchadown
 ....

**Comment by chrisdavenport** [comment]
I have tested this item :white_check_mark: successfully on d6a18f8ff2bce6ccde723ff6d05bffef33b95f10<br /><br />Works perfectly and looks good.  Thanks Nicola. :smile: <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/weblinks/223">issues.joomla.org/weblinks/223</a>.</sub>

**Comment by alikon** [comment]
there is the need to add the lang sing to the joomla-cms repo too ?
https://github.com/joomla/joomla-cms/pull/10895#issuecomment-228271641

**Comment by yvesh** [comment]
@alikon There is a slight css issue for me:

screenshot 2016-06-28 16 44 40

Imo it's not completely related to your PR, even the <hr /> element has that issue. This is a clean Joomla! staging / protostar + weblinks installation.

Everything else works as expected, great work Nicola!

**Comment by alikon** [comment]
@yvesh css is not my "comfort zone"

any help to fix CSS issue ?

**Comment by alikon** [comment]
...it happens only when editor is CodeMirror...

**Comment by yvesh** [comment]
@alikon it also happens without an editor. But you are right with TinyMCE it's fine.. Probably something which should be fixed in core (the row with buttons need to set width: 100% for the editor line, so there is no free space right of it).

We could fix it for weblinks with the following css:

#editor-xtd-buttons {
   width: 100%
}

@chrisdavenport What do you think? Core problem or weblinks?

screenshot 2016-06-29 21 34 38

**Comment by dgt41** [comment]
@alikon @yvesh how about: https://github.com/joomla/joomla-cms/pull/10970

**Comment by alikon** [comment]
maybe @dgt41  can evaluate my dirty fix on https://github.com/joomla/joomla-cms/blob/staging/templates/protostar/css/template.css#L2609
from

```css
.btn-toolbar {
	font-size: 0;
	margin-top: 9px;
	margin-bottom: 9px;
}
```
to
```css
.btn-toolbar {
	font-size: 0;
	margin-top: 9px;
	margin-bottom: 9px;
        width: 100%;
}
```
results is clean now, i think...
capweblilnk

but i don't know if it is a goog fix .....

**Comment by yvesh** [comment]
@alikon looks fine for me, not sure that would cause other issues though. You could also just remove the `pull-left` class the editor-xtd-buttons div has. Not sure why that is there at all?

Is there a case were there is something located right to it on the same line?

// cc @dgt41

**Comment by chrisdavenport** [comment]
Any objections to merging this now?

**Comment by alikon** [comment]
For my knowledge of CSS the issue we have should be fixed on the CMS CSS side
We are waiting for some CSS expert answer which should be the right fix mine or Yves or ...
Evoking @dgt41

**Comment by alikon** [comment]
yes please merge this PR

p.s.
cause i'm unable to find the latest damned branch conflicts :sob:

**Comment by yvesh** [comment]
@alikon i sent you an PR against yours, merging current master changes into it. (Test please)

**Comment by yvesh** [comment]
Okay conflicts resolve. @chrisdavenport you can merge now.

**Comment by alikon** [comment]
thanks 1000  @yvesh

**Comment by chrisdavenport** [comment]
Merging.  Good work everyone. :-)

**Pull request merged**